agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -102,14 +98,19 @@ func TestAwslogsDriver(t *testing.T) {
 		LogGroupName:  aws.String(awslogsLogGroupName),
 		LogStreamName: aws.String(fmt.Sprintf("ecs-functional-tests/awslogs/%s", taskId)),
 	}
-	resp, err := cwlClient.GetLogEvents(params)
-	if err != nil {
-		t.Fatalf("CloudWatchLogs get log failed: %v", err)
-	}
 
-	if len(resp.Events) != 1 {
-		t.Errorf("Get number of log events: %d", len(resp.Events))
+	var resp *cloudwatchlogs.GetLogEventsOutput
+	// The test could fail for timing issue, so retry for 30 seconds to make this test more stable
+	for i := 0; i < 30; i++ {
+		resp, err = cwlClient.GetLogEvents(params)
```

Comment:
Retries exist for all calls.  Might be better to override the `Retryer` instead of doing this here.  I don't see any waiters yet (though that wouldn't be a bad thing to write; it'd just be a jmespath expression).